File: hook_system/client/ClientServer/Models/Package.cs
using System;
using System.Collections.Generic;

namespace ClientServer.Models
{
    public class Package
    {
        public long PackageId { get; set; }
        public long AssignmentId { get; set; }
        public string JobId { get; set; }
        public DateTime EstimatedCompletion { get; set; }
        public string Status { get; set; }
       
        public virtual Assignment Assignment { get; set; }
        public virtual Result Result { get; set; }
        public virtual ICollection<PreviousAssignment> PreviousAssignments { get; set; }
    }
}

File: hook_system/client/ClientServer/ClientApp/src/app/package/package.ts
import { Assignment } from '../course/course';

export class PreviousAssignment {
    assignmentId: number;
    packageId: number;
    assignment: Assignment;
    pack: Package;
}

export class Package {
    assignmentId: number;
    resultId: number;
    estimatedCompletion: Date;
    assignment: Assignment;
    // result: Result;
    previousAssignments: PreviousAssignment[];
}